lib/spack/docs/known_issues.rst

```diff
@@ -0,0 +1,36 @@
+============
+Known Issues
+============
+
+This is a list of known bugs in Spack. It provides ways of getting around these
+problems if you encounter them.
+
+-----------------------------------------------------------------
+Default variants are not taken into account during concretization
+-----------------------------------------------------------------
+
+**Status:** Expected to be fixed in the next release
+
+Current conretization algorithm does not take into account default values
+of variants when adding extra constrains to the spec via CLI. For example
+you may enounter the following error when trying to specify which MPI provider
+to use
+
+.. code-block:: console
+
+   $ spack install hdf5 ^openmpi
+   ==> Error: hdf5 does not depend on openmpi
+
+although the hdf5 package contains
+
+.. code-block:: python
+
+   depends_on('mpi', when='+mpi')
```

Comment:
Also add:
```python
variant('mpi', default=True, description='Enable MPI support')
```
Whether or not the variant is activated by default is important to this bug.